applications/vessel-history/src/features/settings/settings.hooks.ts

```diff
@@ -46,3 +53,82 @@ export const useApplySettingsConnect = () => {
     setSettingOptions,
   }
 }
+
+export const useSettingsRegionsConnect = (
+  section: SettingEventSectionName,
+  settings: SettingsEvents
+) => {
+  const { t } = useTranslation()
+  const { setSettingOptions } = useSettingsConnect()
+
+  const onlyUnique = useCallback((value: Region, index: number, self: Region[]) => {
+    return self.map((item) => item.id).indexOf(value.id) === index
+  }, [])
+
+  const EEZ_REGIONS: Region[] = useSelector(selectEEZs)
+  const RFMOS_REGIONS: Region[] = useSelector(selectRFMOs).filter(onlyUnique)
+  const MPAS_REGIONS: Region[] = useSelector(selectMPAs)
+  const COUNTRIES: Region[] = flags
+
+  const anyOption: MultiSelectOption<string> = useMemo(
+    () => ({
+      ...anyRegion,
+      label: t(`common.${anyRegion.label}` as any, capitalize(anyRegion.label)) as string,
+      disabled: false,
+      tooltip: t(`common.${anyRegion.label}` as any, capitalize(anyRegion.label)) as string,
+    }),
+    [t]
+  )
+
+  const onSelectRegion = useCallback(
+    (selected: MultiSelectOption<string>, currentSelected: MultiSelectOption[], field: string) => {
+      selected === anyOption
+        ? // when ANY is selected the rest are deselected
+          setSettingOptions(section, field, [selected])
+        : // when other than ANY is selected
+          setSettingOptions(section, field, [
+            // then ANY should be deselected
+            ...currentSelected.filter((option) => option !== anyOption),
+            selected,
+          ])
+    },
+    [section, setSettingOptions, anyOption]
+  )
+
+  const getOptions = useCallback(
+    (availableOptions: MultiSelectOption[], field: string, selected?: string | string[]) => {
+      const allOptions = [anyOption, ...availableOptions]
+      const selectedOptions = allOptions.filter((option) => selected?.includes(option.id))
+      const options = [
+        // First display ANY whether it's selected or not
+        anyOption,
```

Comment:
what happen if any is selected, and any other random option? is like all are selected so if any is selected we can avoid display the list, can that help to the performance? is an idea